pkg/handler/auth/check_auth.go

```diff
@@ -0,0 +1,58 @@
+package auth
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+var hlog = logrus.WithField("module", "handler.auth")
+
+const AuthHeader = "Authorization"
+
+func GetUserToken(header http.Header) (string, error) {
+	authValue := header.Get(AuthHeader)
+	if authValue != "" {
+		parts := strings.Split(authValue, "Bearer ")
+		if len(parts) != 2 {
+			return "", errors.New("missing Bearer token in Authorization header")
+		}
+		return parts[1], nil
+	}
+	return "", errors.New("missing Authorization header")
+}
+
+func CheckAuth(ctx context.Context, header http.Header) error {
+	hlog.Debug("Checking authenticated user")
+	token, err := GetUserToken(header)
+	if err != nil {
+		return err
+	}
+	hlog.Debug("Checking auth: token found")
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		return err
+	}
+	config.BearerToken = token
+	config.BearerTokenFile = ""
+
+	client, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return err
+	}
+	hlog.Debug("Checking auth: kube config created")
+
+	_, err = client.CoreV1().Namespaces().List(ctx, v1.ListOptions{})
```

Comment:
what about relying on `client.AuthorizationV1().SelfSubjectAccessReviews()` here ?

Reply:
or just `client.AuthenticationV1().TokenReviews()` if you don't want to check any role

Reply:
Checking if a token is valid by trying to list namespaces with it looks a bit hacky.
As a first quick fix this is fine but we should look for a better solution IMO.

Reply:
I'll check what Julien suggests

Reply:
So, I think for a quick fix we should go ahead with GetNamespaces, bc using SSAR / TokenReviews requires permission from the console plugin to create these objects (so need changes in the operator etc.)

Also, I remember to have seen a similar issue and the `GetNamespaces` solution (call it a hack if you want :) ) was used because working with more k8s distributions (including ones that don't necessarily have RBAC out of the box)

Reply:
maybe just add a TODO comment in that case